### PR TITLE
make update-otel

### LINKED ISSUE #21632: [chore] [CI] Include updating the otelcontribcol ocb manifest in make update-otel target
`make update-otel` should additionally update collector core version in `cmd/otelcontribcol/builder-config.yaml` and run `make genotelcontribcol` to avoid doing it manually every time

### PR DESCRIPTION
**Pull request by led0nk**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>
- Resolves #21632 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

So at first i reordered the `update-otel` command, because there was no sense in first updating the builder-configs (with the use of go.mod file) and then updating `go.mod`-files.

Furthermore since the main `go.mod`-file has moved:
https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/cfd83af71471ae54f6489355c47cac56c5f746e1#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

I changed the source-file to `cmd/otelcontribcol/go.mod`

### CODE REVIEW

**Comment by led0nk** [comment]
@mx-psi could you have a look at this? I think it should solve the problem with `make update-otel`

**Review by mx-psi — APPROVED** [review]
Thanks, I think this should work. I am going to wait until the current release is done to merge this

**Review by codeboten — APPROVED** [review]
thanks @led0nk

**Comment by mx-psi** [comment]
This was a step in the right direction, but it looks like there are still a couple manual steps left. 

For example, `check-module-collector-version` failed on #35158: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10828278572/job/30043431071?pr=35158. One solution is to move `git diff` on that script out, and run the script as part of `make update-otel` before `make gotidy`.

Another cumbersome thing is replacing the `toolchain` directive manually, we can solve that independently

**Comment by mx-psi** [comment]
Filed #35159 and #35160 for this.

**Pull request merged**
